rfcs/0187-sign-taskcluster-yml-remote-references.md

```diff
@@ -0,0 +1,31 @@
+# RFC 187 - Integrity checks for .taskcluster.yml remote references (addendum to #182)
+* Comments: [#187](https://github.com/taskcluster/taskcluster-rfcs/pull/186)
+* Proposed by: @bhearsum
+
+# Summary
+
+Require remotely referenced `.taskcluster.yml` files to be verifiable through a detached signature.
+
+## Motivation
+
+In #182 we discussed and agreed on a mechanism to allow for deduplication of `.taskcluster.yml` files by supporting pointers to a `.taskcluster.yml` file hosted elsewhere. Something we did not discuss there was how to verify that these files have not been tampered with (being overridden, man-in-the-middle, etc.).
+
+This is a crucial detail to iron out before we implement it.
+
+# Details
+
+To accommodate integrity checks, Taskcluster-GitHub will require that any remotely referenced `.taskcluster.yml` files have an associated detached signature which can be verified by a public key that it has been configured with.
+
+Taskcluster-GitHub will allow for zero, one, or many public keys to be configured in its `config.yml`. If zero public keys are configured, remote references are not supported. If one or more keys are configured, any `.taskcluster.yml` that is verifiable by _any_ of the configured keys is permitted to be used. (That is to say: keys are not scoped in any way, nor does the order in which they are listed in the config matter.)
```

Comment:
If I understand correctly, if no config will be provided, github service will not parse `config-from` at all? 
Maybe we can allow non fxci deployments to stay flexible, and use those includes without signatures?

Reply:
That's a good point - Firefox CI requirements should not be dictating other clusters. I suspect SecOps will want the default case to be "disallow", however.

What do you think about having another configuration parameter, something like "require-signed-configs" which defaults to True? (That way any accidental misconfiguration that results in keys not being listed will not degrade security.)

Reply:
Maybe this option could be named as `allow-unsigned-configs: [False]` to make it clear what admin is enabling

Reply:
That sounds reasonable. (Why is it a list though?)

Reply:
I pushed a change to include this -- I ended up calling it `allow-unsigned-remote-references` to avoid confusion between in-repo `.taskcluster.yml` files and external ones. (Both can be thought of as "configs", I think...) I'm happy to change the name if preferred though.